File: blogBackend/src/main/java/com/example/blogbackend/entity/Comment.java
package com.example.blogbackend.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;
import java.time.LocalDateTime;

/**
 * <p>
 * 评论表
 * </p>
 *
 * @author author
 * @since 2024-04-12
 */
@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
@TableName("comments")
public class Comment {

    private static final long serialVersionUID = 1L;

    /**
     * 评论ID
     */
    @TableId(value = "id", type = IdType.AUTO)
    private Integer id;

    /**
     * 关联的文章ID
     */
    private Integer articleId;

    /**
     * 评论用户ID
     */
    private Integer userId;

    /**
     * 父评论ID（用于回复功能）
     */
    private Integer parentId;

    /**
     * 评论内容
     */
    private String content;

    /**
     * 创建时间
     */
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    private LocalDateTime updatedAt;

    /**
     * 网络运营商
     */
    private String networkOperator;

    /**
     * IP属地（省份/城市）
     */
    private String ipLocation;

    /**
     * 浏览器版本
     */
    private String browserVersion;

    /**
     * 操作系统
     */
    private String operatingSystem;
}

File: blogFrontend/src/types/Comment.ts
import { Identifiable, Timestamped, ApiResponse } from './common';

// 评论接口定义
export interface Comment extends Identifiable, Timestamped {
  articleId: number;
  userId: number;
  username: string;
  content: string;
  parentId: number | null;
  avatar?: string;
  likes?: number;
  isLiked?: boolean;
  articleTitle: string;
  replies?: Comment[];

  // 用户设备信息
  networkOperator?: string;
  ipLocation?: string;
  browserVersion?: string;
  operatingSystem?: string;
}

// 评论列表响应

// 评论操作响应
export interface CommentOperationResponse extends ApiResponse<void> { } 